### PR TITLE
fix: xorm.ID() does not support ptr type

### PR DESCRIPTION
**Pull request by Malyue**
#### What this PR does / why we need it:
The new version of `xorm` does not support passing `*uint64` type in the ID method, so the corresponding usage here will be adjusted

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=573348&iterationID=-1&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  fix: xorm.ID() does not support ptr type            |
| 🇨🇳 中文    |    修复了xorm.ID()方法传入指针类型的情况          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sfwn — CHANGES_REQUESTED** [review]
Please add unit test for this scenario.

**Comment by Malyue** [comment]
> Please add unit test for this scenario.

Done.

File: internal/tools/pipeline/providers/reconciler/taskpolicy/provider.go [hunk 1]
```diff
@@ -82,7 +84,10 @@ func (p *provider) resetTask(task *spec.PipelineTask, statuses ...apistructs.Pip
 	}
 	pipeline := statusPipelines[0]
 
-	beforeSuccessTask, err := p.dbClient.GetPipelineTask(pipeline.ParentTaskID)
+	if pipeline.ParentTaskID == nil {
+		return errors.New("Parent Task ID is null")
```
Comment (sfwn) [inline internal/tools/pipeline/providers/reconciler/taskpolicy/provider.go:88]: please see the next few lines about `beforeSuccessTask`.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/providers/reconciler/taskpolicy/provider.go (hunk 1 above)
Comment (Malyue) [reply internal/tools/pipeline/providers/reconciler/taskpolicy/provider.go]: Done.

**Review by sfwn — APPROVED** [review]

**Pull request merged**